src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,159 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private class Node {
+        private T element;
+        private Node next;
+        private Node prev;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        Node newNode = new Node(value);
+        if (index == size) {
+            if (size == 0) {
+                tail = head = newNode;
+            } else {
+                tail.next = newNode;
+                newNode.prev = tail;
+                tail = newNode;
+            }
+            size++;
+            return;
+        }
+        Node findNode = findNode(index);
+        if (findNode != null) {
+            newNode.next = findNode;
+            newNode.prev = findNode.prev;
+            if (findNode.prev != null) {
+                findNode.prev.next = newNode;
+            } else {
+                head = newNode;
+            }
+            findNode.prev = newNode;
+            size++;
+            return;
```

Comment:
too many if-else statements, it can be done more easily

Reply:
without a specific proposal on how to simplify, this is only your value judgment

Reply:
`        Node<T> newNode = new Node<>(value);
        if (head == null) {
            head = tail = newNode;
        } else if (index == 0) {
            newNode.next = head;
            head = newNode;
            head.next.prev = head;
        } else if (index == size) {
            tail.next = newNode;
            tail.next.prev = tail;
            tail = newNode;
        } else {
            Node<T> current = findNode(index - 1);
            newNode.next = current.next;
            newNode.next.prev=newNode;
            current.next=newNode;
            newNode.prev=current;
        }
        size++;
    }`

Reply:
```suggestion
        Node<T> newNode = new Node<>(value);
        if (head == null) {
            head = tail = newNode;
        } else if (index == 0) {
            newNode.next = head;
            head = newNode;
            head.next.prev = newNode;
        } else if (index == size) {
            tail.next = newNode;
            newNode.prev =  tail;
            tail = newNode;
        } else {
            Node<T> current = getNode(index - 1);
            newNode.next = current.next;
            newNode.next.prev=newNode;
            current.next=newNode;
            newNode.prev=current;
        }
        size++;
    }
```

Reply:
something like this

Reply:
```suggestion
 Node<T> newNode = new Node<>(value);
        if (head == null) {
            head = tail = newNode;
        } else if (index == 0) {
            newNode.next = head;
            head = newNode;
            head.next.prev = newNode;
        } else if (index == size) {
            tail.next = newNode;
            newNode.prev =  tail;
            tail = newNode;
        } else {
            Node<T> current = findNode(index - 1);
            newNode.next = current.next;
            newNode.next.prev=newNode;
            current.next=newNode;
            newNode.prev=current;
        }
        size++;
    }
```